### PR TITLE
DOC: Fix EX01 in pandas.DataFrame.idxmax

### PR DESCRIPTION
**Pull request by farhanreynaldo**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Related to #27977. 

```
################################################################################
################################## Validation ##################################
################################################################################

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8029,6 +8029,35 @@ def idxmax(self, axis=0, skipna=True) -> Series:
...
+        ...                    'co2_emission': [37.2, 19.66, 1712]},
+        ...                    index=['Pork', 'Wheat Products', 'Beef'])
+
+        >>> df
+                        consumption  co2_emission
+        Pork                  10.51         37.20
+        Wheat Products       103.11         19.66
+        Beef                  55.48       1712.00
+
+        By default, it returns index for the maximum value in each columns.
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: By default, it returns the index for the maximum value in each column.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -8029,6 +8029,35 @@ def idxmax(self, axis=0, skipna=True) -> Series:
...
+
+        >>> df.idxmax()
+        consumption     Wheat Products
+        co2_emission              Beef
+        dtype: object
+
+        To return index for the maximum value in each rows, use ``axis="columns"``.
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: To return the index for the maximum value in each row, use ``axis="columns"``.

File: pandas/core/frame.py [hunk 3]
```diff
@@ -8029,6 +8029,35 @@ def idxmax(self, axis=0, skipna=True) -> Series:
         Notes
         -----
         This method is the DataFrame version of ``ndarray.argmax``.
+
+        Examples
+        --------
+        Consider dataset containing food consumption in Argentina.
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: Consider a dataset containing food consumption in Argentina.

File: pandas/core/frame.py [hunk 4]
```diff
@@ -8029,6 +8029,35 @@ def idxmax(self, axis=0, skipna=True) -> Series:
...
+
+        >>> df = pd.DataFrame({'consumption': [10.51, 103.11, 55.48],
+        ...                    'co2_emission': [37.2, 19.66, 1712]},
```
Comment (MarcoGorelli) [inline pandas/core/frame.py]: ...                    'co2_emissions': [37.2, 19.66, 1712]},

(this will have to change in the other parts too)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @farhanreynaldo , just suggesting some very minor changes

**Comment by farhanreynaldo** [comment]
Thanks for the input, @MarcoGorelli!

**Review by datapythonista — APPROVED** [review]
Thanks @farhanreynaldo, nice example.

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @farhanreynaldo

**Pull request merged**